source/npm/util.js

```diff
@@ -109,3 +111,57 @@ exports.checkIgnoreStrategy = ({files}) => {
 		`);
 	}
 };
+
+// New files added part of .npmignore or not part of files-attribute in "package.json"
+exports.checkNewFiles = async (newFiles, filesFromFileAttribute) => {
+	const rootDir = pkgDir.sync();
+	const npmignoreExists = fs.existsSync(path.resolve(rootDir, '.npmignore'));
+
+	let result = [];
+
+	if (filesFromFileAttribute) {
+		result = getFilesNotPartOfFileAttribute(filesFromFileAttribute, newFiles);
+	}
+
+	if (npmignoreExists) {
+		result = result.concat(await getFilesIgnoredByDotnpmignore(newFiles));
+	}
+
+	return result;
+};
+
+async function getFilesIgnoredByDotnpmignore(fileList) {
+	if (!Array.isArray(fileList)) {
+		throw new TypeError('expected array, but got {typeof fileList}');
+	}
+
+	const result = [];
+	const whiteList = await ignoreWalker({
+		path: pkgDir.sync(),
+		ignoreFiles: ['.npmignore']
```

Comment:
There are also [files that are ignored by default when `.npmignore` exists](http://npm.github.io/publishing-pkgs-docs/publishing/the-npmignore-file.html).

Reply:
I think the user expects not to see the files, which are ignored by default, even if he added them to the version-control-system. So i will exclude them from the result-set.